src/runtime/providers/string.ts

```diff
@@ -125,6 +124,90 @@ class Substring extends Constraint {
   }
 }
 
+class Find extends Constraint {
+  static AttributeMapping = {
+    "text": 0,
+    "subtext": 1,
+    "case-sensitive": 2,
+    "from": 3,
+  }
+  static ReturnMapping = {
+    "string-position": 0,
```

Comment:
This is going to be the codepoint index right now. I don't think that's a blocker for 6sq, but it probably needs to get an as property before we make it fully public.